tensorboard/plugins/debugger/tensor_helper.py

```diff
@@ -82,25 +82,42 @@ def array_view(array, slicing=None, mapping=None):
   """
 
   dtype = str(array.dtype)
+  # String-type TensorFlow Tensors are represented as object-type arrays in
+  # numpy. We map the type name back to 'string' for clarity.
+  if dtype == 'object':
```

Comment:
Wow, is this the only way to identify a numpy array that represents a TensorFlow string tensor? Do the methods in [this StackOverflow post](https://stackoverflow.com/questions/10790312/numpy-check-array-for-string-data-type) work at all?

Reply:
For a tensorflow string-type tensor, the corresponding numpy array has the dtype "object", i.e., different from a regular string-type array in numpy. This is how tensorflow is and we need to live with that.